### PR TITLE
Cleanly unregister the ingester from consul.

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

File: ingester_lifecycle.go [hunk 1]
```diff
@@ -64,20 +64,21 @@ func RegisterIngester(consulClient ConsulClient, listenPort, numTokens int) (*In
 	return r, nil
 }
 
-func (r *IngesterRegistration) updateLoop() error {
+func (r *IngesterRegistration) updateLoop() {
 	defer r.wait.Done()
 	ticker := time.NewTicker(1 * time.Second)
```
Comment (paulbellamy) [inline ingester_lifecycle.go:69]: Probably want a:

```
defer ticker.Stop()
```

to avoid leaking the goroutine.

File: ingester_lifecycle.go [hunk 2]
```diff
@@ -64,20 +64,21 @@ func RegisterIngester(consulClient ConsulClient, listenPort, numTokens int) (*In
...
+			if err := r.consul.PutBytes(r.id, r.desc); err != nil {
 				log.Errorf("Failed to write to consul, sleeping: %v", err)
+			} else {
+				return
 			}
 		case <-r.quit:
 			ticker.Stop()
+			return
 		}
 	}
```
Comment (paulbellamy) [inline ingester_lifecycle.go:83]: Is the logic you're trying to do: call PutBytes, every second until it succeeds, or we quit?

File: ingester_lifecycle.go (hunk 1 above)
Comment (tomwilkie) [reply ingester_lifecycle.go:69]: +1

File: ingester_lifecycle.go (hunk 2 above)
Comment (tomwilkie) [reply ingester_lifecycle.go:83]: until it succeeds.

Comment (tomwilkie) [reply ingester_lifecycle.go:83]: or is stopped.

**Comment by paulbellamy** [comment]
LGTM

**Pull request merged**